dot-calling-internals.ipynb

```diff
@@ -0,0 +1,2178 @@
+{
```

Comment:
<p>These numbers of clustered pixels are crazy! The CLI call:</p><p><code>cooltools call-dots&nbsp;-p 4&nbsp;--max-loci-separation 1000000&nbsp;--tile-size 1000000&nbsp;--dots-clustering-radius 21000&nbsp;--fdr 0.1&nbsp;--verbose&nbsp;-o xxx.bedpe&nbsp;--regions hg38_arms.bed&nbsp;./test.mcool::/resolutions/10000&nbsp;test.10kb.cis.exp.tsv</code></p><p>results in 12500 filtered pixels that are clustered into 682 final centroids. </p><p> </p><p>I guess it is a problem with the parameters in the notebook that are different from that in CLI... </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open2c/open2c_examples/pull/8/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='51'/>

Reply:
Sash, I wonder if it related to this: https://github.com/open2c/cooltools/pull/228/files#r608106837 - are they only from `chr1`

haven't checked it carefully yet - sorry

Reply:
probably not ...

Reply:
I get ~2'300 centroids from this 12'500 dots in `xxx.bedpe.postproc` and it is reasonable  (2,000 dots *20 chroms ~40'000 genome-wide sounds about right for microC).

in the notebook however - if found a problem with clustering - if you feed a dataframe with messed up index into clustering - it yields incorrect output (more messed up indexing and duplications) - need to investigate that

it's different in cli though - still maybe worth resetting index.

Reply:
Seems to be solved by sorting and resetting the index in dataframes.

Reply:
yes ! nice !